application-task-manager-ui/src/main/resources/TaskManager/Code/TaskReportResultPage.xml

```diff
@@ -37,123 +37,42 @@
+  #set ($list = [])
+  #foreach ($tag in $request.getParameterValues('tasktag'))
+    #set ($discard = $list.add($tag))
   #end
-  #set($discard = $map.put('offset', $mathtool.add($offset, 1)))
-##
-##
-  #set($rows = [])
-  #set($lastDoc = '')
-  #set($lastObjList = '')
-  #foreach($item in $items)
-    #set($rowDocRef = $item.documentReference)
-    #set($objId = $item.id)
-    #set($rowDoc = $xwiki.getDocument($rowDocRef))
-    #set($row = {})
-    #if ($rowDocRef)
-      #set ($row = {
-        'doc_title': $rowDoc.translatedDocument.plainTitle,
-        'doc_viewable': true,
-        'doc_fullName': $services.model.serialize($rowDocRef, 'local'),
-        'doc_space': $services.model.serialize($rowDocRef.parent, 'local'),
-        'doc_url': $xwiki.getURL($rowDocRef),
-        'doc_space_url': $xwiki.getURL($rowDocRef.parent),
-        'doc_wiki': $rowDocRef.wikiReference.name,
-        'doc_wiki_url': $xwiki.getURL($rowDocRef.wikiReference),
-        'doc_hasadmin': $xwiki.hasAdminRights(),
-        'doc_hasedit': $services.security.authorization.hasAccess('edit', $rowDocRef),
-        'doc_hasdelete': $services.security.authorization.hasAccess('delete', $rowDocRef),
-        'doc_edit_url': $xwiki.getURL($rowDocRef, 'edit'),
-        'doc_copy_url': $xwiki.getURL($rowDocRef, 'view', 'xpage=copy'),
-        'doc_delete_url': $xwiki.getURL($rowDocRef, 'delete'),
-        'doc_rename_url': $xwiki.getURL($rowDocRef, 'view', 'xpage=rename&amp;step=1')
-      })
-      #set($discard = $row.put('description', $rowDoc.getRenderedContentRestricted($item.description, $rowDoc.syntax)))
-      #set($discard = $row.put('createDate', $datetool.format($services.taskmanager.configuration.displayDateFormat, $item.createDate)))
-      #set($discard = $row.put('deadline', $datetool.format($services.taskmanager.configuration.displayDateFormat, $item.deadline)))
-      #set($discard = $row.put('completeDate', $datetool.format($services.taskmanager.configuration.displayDateFormat, $item.completeDate)))
-      #set($discard = $row.put('completed', $services.localization.render("taskmanager.livetable.completed.${item.completed}")))
-      #addUserDisplay($row 'creator' $item.creator)
-      #addUserDisplay($row 'assignees' $item.assignees)
-    #else
-      #set($discard = $row.put('description', $services.localization.render('taskmanager.livetable.noRights')))
+  #set ($discard = $params.put("taskTagList", $list))
+#end
+
+#if ("$!request.space" != '')
+  #foreach($space in $request.getParameterValues('space'))
+    #if ($foreach.count > 1)
+      #set ($spaceWhere = "${spaceWhere} OR ")
     #end
-    #set($discard = $rows.add($row))
+    #set ($paramVal = $services.query.parameter().literal("${space}.").anyChars())
+    #set ($spaceWhere = "${spaceWhere}doc.fullName like '${space}.%'")
+    ##set ($discard = $params.put("docspace${foreach.count}", $paramVal))
```

Comment:
I encountered a very peculiar issue here. Initially, I was setting the `docspace${foreach.count}` as `$services.query.parameter().literal("${space}.").anyChars()` and it worked really well. I was able to filter by spaces, I was able to filter by tags but when I had to filter by both of them at the same time, I got a strange error message stating that the `IN` keyword (mentioned above) was unexpected. I ran the resulting query in a separate page and it worked correctly. I also bound the same two parameters to the query and it also worked fine. 

The issue persisted. I observed the following phenomenons: if I didn't set either of the `taskTagList` or `docspace${foreach.count}` it worked fine, even if i filtered for both for spaces and tags. By not setting, I mean either using this syntax `${spaceWhere}doc.fullName like '${space}.%'` or hard-coding a list instead of the `taskTagList`.

This temporary solution is an unsafe one and needs something better.